image-hd.c

```diff
@@ -261,14 +262,14 @@ static int hdimage_insert_gpt(struct image *image, struct list_head *partitions)
 	header.backup_lba = htole64(image->size/512 - 1);
 	header.last_usable_lba = htole64(image->size/512 - 1 - GPT_SECTORS);
 	uuid_parse(hd->disk_uuid, header.disk_uuid);
-	header.starting_lba = htole64(2);
+	header.starting_lba = htole64(hd->gpt_location/512);
```

Comment:
> If that's not what you mean, then could you clarify the problem? Not sure I understand.

They are absolute. But the `starting_lba` in the secondary header should point to the secondary table. This is already incorrect in the existing code, I just noticed this while reading your patch.

Reply:
Putting the secondary table right after the first one makes little sense. But maybe we could just skip it? What happens if `backup_lba` is just set to `1`, so that it points to the primary table?